### ISSUE TITLE
Merge and overwrite resources

### ISSUE THREAD

**Issue by RaptorCZ** [i18next/i18next#331]
Hi, is there any way how to merge resources with "overwrite" option?

Let's say I have static resource string in json files, but I have service returning overwrites and new values. So I need to init framework with static files then merge values. If string doesn't exists - add it. Otherwise replace old value with new one.

My translation.json files are for example:
```javascript
 {
    "BaseResource":
    {
        "AccessDenied": "Acces denied!",
        "Action": "Action",
        "ActionName": "Action "
    },
    
    "Storage":
    {
        "Advertising": "Advertising",
        "All": "-------All-------",
        "AllItems": "All",
        "AutoOwn": "Auto Own"
    }
}
```
Service returns the same JSON structure.

**Comment by iandarlingmnetics** [comment]
I've not been able to run the tests on that commit so far (although I have added one), hence there's not a pull request yet.

**Comment by jamuhl** [comment]
cool...looks like this lands soon :+1:

**Comment by etreworgy** [comment]
There actually is a bug here in the existing _deepExtend (even if you don't intend to allow overwriting).  The function just iterates over 'props in source', and if you happen to reach a leaf of either target or source translation objects you'll get unintended behavior that ultimately results in an error.

If the target hits a leaf string then the '(prop in target)' test fails with an error because JS doesn't let you use 'in' with strings.  If the source happens to hit a leaf string first then the 'for(var prop in source)' starts iterating over the characters and string prototype functions such as 'hashCode'.

Both cases are not expected or documented behavior.  There should be a simple string test on source[prop] and target[prop] to prevent this from happening and then either overwrite or ignore depending on the desired behavior.

**Comment by jamuhl** [comment]
landed in v1.10.3

**Comment by kdydesign** [comment]
Hi, I needed the same option.
I corrected the " ResourceStore.prototype.getResource" function slightly.

```javascript
ResourceStore.prototype.getResource = function getResource(lng, ns, key) {
      var options = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {};
      var self = this;
      var keySeparator = options.keySeparator || this.options.keySeparator;
      var originResource;
      if (keySeparator === undefined) keySeparator = '.';

      var path = [lng, ($.isArray(ns) ? ns[0] : ns)];
      if (key && typeof key !== 'string') path = path.concat(key);
      if (key && typeof key === 'string') path = path.concat(keySeparator ? key.split(keySeparator) : key);

      if (lng.indexOf('.') > -1) {
           path = lng.split('.');
      }

      originResource = getPath(this.data, path);

      //add
      if ($.isArray(ns)) {
           $.each(ns, function (idx, value) {
               var path = [lng, value];

              $.extend(true, originResource, getPath(self.data, path));
          })
      }

      return originResource;
};
```
use it.

```javascript
i18next.getResource(lang, ["ns1", "ns2"]);
```
